### DISCUSSION TITLE
Getting the number of lines which are in a paragraph, especially when the text in the paragraph is wrapped.

### DISCUSSION THREAD

**Discussion by supreetsingh10** [ratatui/ratatui#362] [Ideas]
I have been working on a tui application on my own, I have been going through the code of ratatui and I have would LineWrapper and WordWrapper, but I did not find any direct way to get this information from the library. 

Do you think this would be a good addition to the functionality? I can try to work on this if the community thinks this is relevant.

If there is already a way to get the count, then what is it?

**Comment by joshka** [comment]
There's some work in this direction already happening.
The wordwrap implementation in the reflow module was extended to add some more features and has grown to be difficult to maintain due to the complexity. This spawned an effort to clean it up, which is stalled currently. I wrote an RFC to help highlight that wrapping is one of those tasks that seems simple but actually has a lot of complexity underneath the tip of the iceberg:
- https://github.com/ratatui-org/ratatui/issues/293

**Reply by supreetsingh10** [reply]
Loved the RFC. Amazing work by you guys. I will look for other places in Ratatui where I can make some code contributions. If there is something I can work on, I would be happy to help.

**Comment by a-kenji** [comment]
The [`inner`](https://docs.rs/ratatui/0.22.0/ratatui/widgets/block/struct.Block.html#method.inner) method of a `Block` does something similar. Maybe a design based upon that example can be sufficient?

**Reply by supreetsingh10** [reply]
Thank you I will look into it, I have coded something for my use case, it will need some fine-tuning so this will be helpful.

**Comment by joshka** [comment]
If you need something for your app sooner, take a look at the implementation in https://github.com/joshka/tui-prompts/blob/main/src/text_prompt.rs for some inspiration. It's missing some of the necessary parts (like handling more than 1 Line and handling multi-width characters), but could be helpful as a starting point. I've also been playing a bit with some ideas using the `textwrap` crate. Depending on your use case that might be useful to look at and just avoid the need to worry about the wrapping in ratatui altogether.

**Reply by supreetsingh10** [reply]
Thank you for the references, I will look into these.
